core/src/main/java/com/aem/demo/core/models/NewsCardModel.java

```diff
@@ -62,44 +68,35 @@ public String getFullPath() {
     }
 
 
-    public int  getLikes() {
-        int likes = 0;
 
+    @PostConstruct
+    private void init() {
         String rootPath = resource.getPath();
-
-        resource = resourceResolver.getResource(rootPath);
-        ModifiableValueMap properties = resource != null ? resource.adaptTo(ModifiableValueMap.class) : null;
-
-        if (properties != null) {
-            likes = resource.adaptTo(ValueMap.class).get("likes", 0);
-            properties.put("likes", likes + 1);
-        }
-        try {
-            resourceResolver.commit();
-        } catch (PersistenceException e) {
-            log.info(e.getMessage());
+        String dislikeParam = servletRequest.getParameter("dislike");
+        String likeParam = servletRequest.getParameter("like");
+        if (likeParam != null) {
+            likes = incrementLikeDislike(likeParam, rootPath);
+        } else if (dislikeParam != null) {
+            dislikes = incrementLikeDislike(dislikeParam, rootPath);
         }
-        return likes;
-    }
 
-    public int  getDisLikes() {
-        String dislikeParameter = servletRequest.getParameter("dislikeId");
-        int dislikes = 0;
+    }
 
-        String rootPath = resource.getPath();
 
+    private int incrementLikeDislike(String property, String rootPath) {
+        int incrementProperty = 0;
         resource = resourceResolver.getResource(rootPath);
         ModifiableValueMap properties = resource != null ? resource.adaptTo(ModifiableValueMap.class) : null;
-
-        if (properties != null && dislikeParameter.equals("false")) {
-            dislikes = resource.adaptTo(ValueMap.class).get("dislikes", 0);
-            properties.put("dislikes", dislikes + 1);
+        if (properties != null) {
+            incrementProperty = properties.get(property, 0);
+            properties.put(property, incrementProperty + 1);
         }
         try {
             resourceResolver.commit();
         } catch (PersistenceException e) {
-            log.info(e.getMessage());
+            log.error("Exception occurred while incrementing like or dislike " + e.getMessage());
         }
-        return dislikes;
+        return incrementProperty;
```

Comment:
Please be aware that you return the not incremented property. In line 92 you store the `incrementProperty + 1` but then you actually return the original value!
Therefore, the `likes` or `dislikes` field will not be actually incremented if the eyes of a user.